stl/inc/syncstream

```diff
@@ -0,0 +1,312 @@
+            return _Traits::eof();
+
+        const size_type _New_capacity  = _Calculate_growth(_Buf_size, _Buf_size + 1, _Max_allocation);
+        const pointer _Old_ptr         = pointer_traits<pointer>::pointer_to(*streambuf_type::pbase());
+        const size_type _Old_data_size = _Get_data_size();
+
+        const pointer _New_ptr = _Unfancy(_Al.allocate(_New_capacity));
+        _Traits::copy(_New_ptr, _Old_ptr, _Old_data_size);
+
+        streambuf_type::setp(_New_ptr, _New_ptr + _Old_data_size, _New_ptr + _New_capacity);
+        streambuf_type::sputc(_Current_elem);
+
+        return _Current_elem;
+    }
+
+private:
+    enum { // constant for minimum buffer size
+        _MINSIZE = 32
+    };
+
+    void _Init() {
+        const pointer _New_ptr = _Unfancy(_Al.allocate(_MINSIZE));
+        streambuf_type::setp(_New_ptr, _New_ptr + _MINSIZE);
+    }
+
+    void _Tidy() noexcept {
+        const size_type _Buf_size = _Get_buffer_size();
+        _Al.deallocate(pointer_traits<pointer>::pointer_to(*streambuf_type::pbase()), _Buf_size);
+
+        streambuf_type::setp(nullptr, nullptr, nullptr);
+        if (_Wrapped != nullptr) {
+            _Delete_mutex_for_instance_or_decrement(static_cast<void*>(_Wrapped));
+            _Wrapped = nullptr;
+        }
+    }
+
+    void _Assign_rv(basic_syncbuf&& _Right) {
+        if (this != _STD addressof(_Right)) {
+            _Tidy();
+            this->_Swap_except_al(_Right);
+        }
+    }
+
+    void _Move_assign(basic_syncbuf&& _Right, _Equal_allocators) noexcept {
+        _Tidy();
+        _Pocma(_Getal(), _Right._Getal());
+        this->_Swap_except_al(_Right);
+    }
+
+    void _Move_assign(basic_syncbuf&& _Right, _Propagate_allocators) noexcept {
+        _Tidy();
+        _Pocma(_Getal(), _Right._Getal());
+        this->_Swap_except_al(_Right);
+    }
+
+    void _Move_assign(basic_syncbuf&& _Right, _No_propagate_allocators) {
+        if (_Getal() == _Right._Getal()) {
+            _Move_assign(_Right, _Equal_allocators{});
+        } else {
+            const size_type _Right_Buf_size  = _Right._Get_buffer_size();
+            const size_type _Right_Data_size = _Right._Get_data_size();
+
+            const pointer _New_ptr = _Unfancy(_Al.allocate(_Right_Buf_size));
+            _Traits::copy(_New_ptr, pointer_traits<pointer>::pointer_to(*_Right.pbase()), _Right_Data_size);
+
+            streambuf_type::setp(_New_ptr, _New_ptr + _Right_Data_size, _New_ptr + _Right_Buf_size);
+            streambuf_type::_Plocale = _STD move(_Right._Plocale);
+
+            _Right._Tidy();
+        }
+    }
+
+    void _Swap_except_al(basic_syncbuf& _Right) {
+        streambuf_type::swap(_Right);
+        _STD swap(_Emit_on_sync, _Right._Emit_on_sync);
+        _STD swap(_Record_sync, _Right._Record_sync);
+        _STD swap(_Wrapped, _Right._Wrapped);
+    }
+
+    bool _Emit() noexcept {
+        _TRY_BEGIN
+        return emit();
+        _CATCH_ALL
+        _CATCH_END
+    }

```

Comment:
If this throws an exception, it will fall off of the end of the function without returning a `bool`.